### DISCUSSION TITLE
hiredis_ssl missing

### DISCUSSION THREAD

**Discussion by mike1821** [redis/hiredis#1203] [Q&A] [answered]
Hello,

I am trying to build hiredis using USE_SSL=1 flag, but it seems no libhiredis_ssl.so is compiled during make. Am I missing something here?

**Comment by michael-grunder** [comment]
That should be all you need.  Can you provide more context about exactly how you're building hiredis and on which system, etc?

**Reply by mike1821** [reply]
Hi, thanks for your reply. I am trying to build in a centos 7 container. I have installed opensll-devel l, gcc, etc. 

I created a build directory and I am executing inside it

cmake ../
make USE_SSL=1 
make install USE_SSL=1

It is very strange since, if I just execute a make and make install command from the parent source directory, the SSL library builds correctly.

However, when I try to use it to compile redis-plus-plus it fails in find_package step.

**Comment by michael-grunder** [comment, marked as answer]
Aha!

If you're using CMake to build hiredis you would do it slightly differently.

CMakeLists.txt defines these options:

```bash
OPTION(BUILD_SHARED_LIBS "Build shared libraries" ON)
OPTION(ENABLE_SSL "Build hiredis_ssl for SSL support" OFF)
OPTION(DISABLE_TESTS "If tests should be compiled or not" OFF)
OPTION(ENABLE_SSL_TESTS "Should we test SSL connections" OFF)
OPTION(ENABLE_EXAMPLES "Enable building hiredis examples" OFF)
OPTION(ENABLE_ASYNC_TESTS "Should we run all asynchronous API tests" OFF)
```

So to build with SSL enabled you would so something like this:

```bash
$ cmake .. -DENABLE_SSL=ON
$ make
```
You can similarly enable/disable the other defined options by passing `-D<option_name>=<ON/OFF>`.

**Reply by mike1821** [reply]
Much appreciated! Thank you very much
